specification/developersigning/DeveloperSigning/main.tsp

```diff
@@ -0,0 +1,196 @@
+using Azure.Core;
+
+@useAuth(
+  OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit,
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+      scopes: ["https://codesigning.azure.net/.default"],
+    }
+  ]>
+)
+@service({
+  title: "Azure Developer Signing",
+  summary: "Azure Developer Signing is a service that provides managed artifact signing for all.",
+})
+@versioned(Azure.Developer.Signing.Versions)
+@server(
+  "https://{region}.codesigning.azure.net/",
+  "Sign and retrieve relevant information pertaining signing process.",
+  {
+    @doc("The Azure region wherein requests for signing will be sent.")
+    region: string,
+  }
+)
+@doc("Azure Developer Signing is a service that provides managed artifact signing for all.")
+namespace Azure.Developer.Signing;
+
+@doc("Versions info.")
+enum Versions {
+  @doc("The 2023-06-15-preview version.")
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  v2023_06_15_preview: "2023-06-15-preview",
+}
+
+interface CertificateProfileOperations {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "This is a custom operation status endpoint."
+  @summary("Gets the status of a signing operation.")
+  @doc("This status operation requires that a Sign request has been submitted and the operationId is known.")
+  @route("/codesigningaccounts/{accountName}/certificateprofiles/{certificateProfile}/sign")
+  getSigningStatus is Foundations.GetOperationStatus<
+    SigningOptions,
+    SignResult
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/rpc-operation-request-body" "no body included, bytesbody is the response object."
+  #suppress "@azure-tools/typespec-azure-core/no-rpc-path-params" "Service needs rpc paths to follow pattern of other services."
+  @summary("Gets the signing root certificate on the certificate chain for that account and profile name.")
+  @doc("The root certificate is generated as part of the initial account creation and it is used to sign the bits for the profile provided.")
+  @route("/codesigningaccounts/{accountName}/certificateprofiles/{certificateProfile}/sign/rootcert")
+  getSignRootCertificate is RpcOperation<
+    SigningOptions,
+    BytesBody<"application/x-x509-ca-cert">
+  >;
+
+  @summary("Gets a list of extended key usage object identifiers that are allowed for this account and profile combination.")
+  @doc("The list of extended key usages are used to determine the purpose of the certificate usage as part of the signing operation.")
+  listExtendedKeyUsages is StandardResourceOperations.ResourceList<ExtendedKeyUsage>;
+
+  @summary("Submit a sign operation.")
+  @doc("Submit a sign operation under the created account and profile name provided.")
+  @pollingOperation(CertificateProfileOperations.getSigningStatus)
+  sign is StandardResourceOperations.LongRunningResourceAction<
+    CertificateProfileName,
+    SigningPayloadOptions,
+    SignResult,
+    SigningRequestHeadersTraits
```

Comment:
You are missing the error response here.
```suggestion
    SignResult,
    Foundations.ErrorResponse,
    SigningRequestHeadersTraits
```

Reply:
Thanks, this is fixed on the PR.